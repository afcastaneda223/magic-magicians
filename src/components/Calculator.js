import React from 'react';

export class Calculator extends React.Component {
  btnCLass = {
    grey: 'btn btn-light col py-4',
    orange: 'btn btn-warning col py-4',
    zero: 'btn btn-light col-6 py-4',
  }

  render() {
    return (
      <div className="container" id="CalculatorContainer">
        <input type="number" id="InputNumber" className="form-control fs-1 text-white text-end py-4" defaultValue="0" />
        <div className="row row-cols-4 container m-0 p-0">
          <button type="button" className={this.btnCLass.grey}>AC</button>
          <button type="button" className={this.btnCLass.grey}>+/-</button>
          <button type="button" className={this.btnCLass.grey}>%</button>
          <button type="button" className={this.btnCLass.orange}>รท</button>
          <button type="button" className={this.btnCLass.grey}>7</button>
          <button type="button" className={this.btnCLass.grey}>8</button>
          <button type="button" className={this.btnCLass.grey}>9</button>
          <button type="button" className={this.btnCLass.orange}>x</button>
          <button type="button" className={this.btnCLass.grey}>4</button>
          <button type="button" className={this.btnCLass.grey}>5</button>
          <button type="button" className={this.btnCLass.grey}>6</button>
          <button type="button" className={this.btnCLass.orange}>-</button>
          <button type="button" className={this.btnCLass.grey}>1</button>
          <button type="button" className={this.btnCLass.grey}>2</button>
          <button type="button" className={this.btnCLass.grey}>3</button>
          <button type="button" className={this.btnCLass.orange}>+</button>
          <button type="button" className={this.btnCLass.zero}>0</button>
          <button type="button" className={this.btnCLass.grey}>.</button>
          <button type="button" className={this.btnCLass.orange}>=</button>
        </div>
      </div>
    );
  }
}

export default Calculator;
