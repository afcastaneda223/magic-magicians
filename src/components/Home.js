import React from 'react';

function Home() {
  return (
    <div className="position-absolute top-50 start-50 translate-middle">
      <h1>Welcome to my Calculator</h1>
      <h5 className="mt-4">
        {' '}
        Lorem ipsum dolor sit amet consectetur adipisicing elit.
        Nostrum amet a maxime perferendis praesentium fugit illum
        quam saepe, omnis nisi delectus enim dolore unde nemo ab
        tempora quod earum minus iusto laborum! Omnis laudantium ex
        {' '}
      </h5>
    </div>
  );
}

export default Home;
