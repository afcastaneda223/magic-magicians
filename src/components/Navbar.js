import React from 'react';
import { Link } from 'react-router-dom';

function Navbar() {
  return (
    <nav className="navbar navbar-expand-lg navbar-dark bg-dark mb-5 sticky-top">
      <div className="container-fluid">
        <Link to="/"><button type="button" className="btn btn-dark"><h3>Math Magicians</h3></button></Link>
        <button
          className="navbar-toggler"
          type="button"
          data-bs-toggle="collapse"
          data-bs-target="#navbarSupportedContent"
          aria-controls="navbarSupportedContent"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon" />
        </button>
        <div className="collapse navbar-collapse" id="navbarSupportedContent">
          <ul className="navbar-nav me-auto mb-2 mb-lg-0 mx-auto pe-5">
            <li className="nav-item m-2">
              <Link to="/"><button type="button" className="btn btn-dark"><h5>Home</h5></button></Link>
            </li>
            <li className="nav-item m-2">
              <Link to="/calculator"><button type="button" className="btn btn-dark"><h5>Calculator</h5></button></Link>
            </li>
            <li className="nav-item m-2">
              <Link to="/quote"><button type="button" className="btn btn-dark"><h5>Quote</h5></button></Link>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  );
}

export default Navbar;
