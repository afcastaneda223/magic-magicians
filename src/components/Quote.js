import React from 'react';

function Quote() {
  return (
    <div className="position-absolute top-50 start-50 translate-middle">
      <h1>Quote</h1>
      <h5 className="mt-5">
        Mathematics is not about numbers, equations, computations,
        <br />
        or algorithms: it is about understanding. -William Paul Thurston
      </h5>
    </div>
  );
}

export default Quote;
