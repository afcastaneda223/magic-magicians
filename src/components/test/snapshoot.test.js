/* eslint-disable no-unused-vars */
import TestRenderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Calculator from '../Calculator';
import Quote from '../Quote';
import Home from '../Home';
import Navbar from '../Navbar';

describe('Navbar', () => {
  it('Renders the Navbar', () => {
    const calculator = TestRenderer.create(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    ).toJSON();
    expect(calculator).toMatchSnapshot();
  });
});

describe('Home', () => {
  it('Renders the Home', () => {
    const home = TestRenderer.create(
      <Home />,
    ).toJSON();
    expect(home).toMatchSnapshot();
  });
});

describe('Calculator', () => {
  it('Renders the Calculator', () => {
    const calculator = TestRenderer.create(
      <Calculator />,
    ).toJSON();
    expect(calculator).toMatchSnapshot();
  });
});

describe('Quote', () => {
  it('Renders the Quote', () => {
    const quote = TestRenderer.create(
      <Quote />,
    ).toJSON();
    expect(quote).toMatchSnapshot();
  });
});
