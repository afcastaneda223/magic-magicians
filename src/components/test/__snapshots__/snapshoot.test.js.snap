// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Renders the Calculator 1`] = `
<div
  className="row"
>
  <div
    className="col-4"
  >
    <h1>
      Lets do some math!
    </h1>
  </div>
  <div
    className="col-8"
  >
    <input
      className="form-control fs-1 text-white text-end py-lg-4 py-sm-3"
      id="InputNumber"
      type="text"
    />
    <div
      className="row row-cols-4 container m-0 p-0"
    >
      <button
        className="btn btn-light col py-lg-5 py-sm-4"
        name="AC"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn btn-light col py-lg-5 py-sm-4"
        name="+/-"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn btn-light col py-lg-5 py-sm-4"
        name="%"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn btn-warning col py-lg-5 py-sm-4"
        name="รท"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="btn btn-light col py-lg-5 py-sm-4"
        name="7"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn btn-light col py-lg-5 py-sm-4"
        name="8"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn btn-light col py-lg-5 py-sm-4"
        name="9"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn btn-warning col py-lg-5 py-sm-4"
        name="x"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="btn btn-light col py-lg-5 py-sm-4"
        name="4"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn btn-light col py-lg-5 py-sm-4"
        name="5"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn btn-light col py-lg-5 py-sm-4"
        name="6"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn btn-warning col py-lg-5 py-sm-4"
        name="-"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="btn btn-light col py-lg-5 py-sm-4"
        name="1"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn btn-light col py-lg-5 py-sm-4"
        name="2"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn btn-light col py-lg-5 py-sm-4"
        name="3"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn btn-warning col py-lg-5 py-sm-4"
        name="+"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="btn btn-light col-6 py-lg-5 py-sm-4"
        name="0"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn btn-light col py-lg-5 py-sm-4"
        name="."
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn btn-warning col py-lg-5 py-sm-4"
        name="="
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`Home Renders the Home 1`] = `
<div
  className="position-absolute top-50 start-50 translate-middle"
>
  <h1>
    Welcome to my Calculator
  </h1>
  <h5
    className="mt-4"
  >
     
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Nostrum amet a maxime perferendis praesentium fugit illum quam saepe, omnis nisi delectus enim dolore unde nemo ab tempora quod earum minus iusto laborum! Omnis laudantium ex
     
  </h5>
</div>
`;

exports[`Navbar Renders the Navbar 1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark mb-5 sticky-top"
  >
    <div
      className="container-fluid"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <button
          className="btn btn-dark"
          type="button"
        >
          <h1>
            Math Magicians
          </h1>
        </button>
      </a>
      <button
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-bs-target="#navbarSupportedContent"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarSupportedContent"
      >
        <ul
          className="navbar-nav me-auto mb-2 mb-lg-0 mx-auto pe-5"
        >
          <li
            className="nav-item m-2"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              <button
                className="btn btn-dark"
                type="button"
              >
                <h3>
                  Home
                </h3>
              </button>
            </a>
          </li>
          <li
            className="nav-item m-2"
          >
            <a
              href="/calculator"
              onClick={[Function]}
            >
              <button
                className="btn btn-dark"
                type="button"
              >
                <h3>
                  Calculator
                </h3>
              </button>
            </a>
          </li>
          <li
            className="nav-item m-2"
          >
            <a
              href="/quote"
              onClick={[Function]}
            >
              <button
                className="btn btn-dark"
                type="button"
              >
                <h3>
                  Quote
                </h3>
              </button>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>,
  ",",
]
`;

exports[`Quote Renders the Quote 1`] = `
<div
  className="position-absolute top-50 start-50 translate-middle"
>
  <h1>
    Quote
  </h1>
  <h5
    className="mt-4"
  >
    Mathematics is not about numbers, equations, computations,
    <br />
    or algorithms: it is about understanding. -William Paul Thurston
  </h5>
</div>
`;
